### PR TITLE
[tools] Update to babel 7 and add jest

### PR DESCRIPTION
**Pull request by kobawan**
### Description

Introduced Jest to start moving away from Mocha tests.
Upgraded babel to 7 since Jest version 24 only supports babel 7.

### Issue

https://github.com/dailymotion/vast-client-js/issues/278

### Type
- [ ] Breaking change
- [x] Enhancement
- [ ] Fix
- [ ] Documentation
- [x] Tooling

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  // collectCoverage: false,
+
+  // An array of glob patterns indicating a set of files for which coverage information should be collected
+  // collectCoverageFrom: null,
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: 'coverage'
```
Comment (eliamaino-fp) [inline jest.config.js]: do we need this?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+  // An array of file extensions your modules use
+  // moduleFileExtensions: [
+  //   "js",
+  //   "json",
+  //   "jsx",
+  //   "ts",
+  //   "tsx",
+  //   "node"
+  // ],
```
Comment (eliamaino-fp) [inline jest.config.js]: no need to set this?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  // testEnvironmentOptions: {},
+
+  // Adds a location field to test results
+  // testLocationInResults: false,
+
+  // The glob patterns Jest uses to detect test files
+  // testMatch: [
+  //   "**/__tests__/**/*.[jt]s?(x)",
+  //   "**/?(*.)+(spec|test).[tj]s?(x)"
+  // ],
```
Comment (eliamaino-fp) [inline jest.config.js]: do we need this?

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  // resetMocks: false,
+
+  // Reset the module registry before running each individual test
+  // resetModules: false,
+
+  // A path to a custom resolver
+  // resolver: null,
+
+  // Automatically restore mock state between every test
+  // restoreMocks: false,
```
Comment (eliamaino-fp) [inline jest.config.js]: what's the difference with the clearMocks option?

File: jest.config.js (hunk 2 above)
Comment (kobawan) [reply jest.config.js]: these are the default values already, we only need to uncomment them if we need to make changes

File: jest.config.js (hunk 3 above)
Comment (kobawan) [reply jest.config.js]: this is already working as default

File: jest.config.js (hunk 4 above)
Comment (kobawan) [reply jest.config.js]: clear mocks resets the mock.calls and mock.instances whereas restore mocks clears the mocks and removes the mock implementation itself, so if you mocked a return value, clear mock wont remove it and reset mock will. https://jestjs.io/docs/en/mock-function-api.html

File: jest.config.js (hunk 1 above)
Comment (kobawan) [reply jest.config.js]: I can comment it back

**Comment by kobawan** [comment]
@eliamaino-fp sure the comments arent needed, I can just leave link to documentation

**Comment by kobawan** [comment]
@eliamaino-fp  @babel/preset-env targets es2015 by default. In the docs: https://babeljs.io/docs/en/babel-preset-env
`Sidenote, if no targets are specified, @babel/preset-env will transform all ECMAScript 2015+ code by default.`

**Pull request merged**